### DISCUSSION TITLE
ipv4、ipv6双栈下，添加ipv4优先选项

### DISCUSSION THREAD

**Discussion by snaking999** [pymumu/smartdns#1720] [Ideas]
因为现在运营商的ipv6非常不稳定，时通时不通，是不是可以添加一个ipv4优先功能选项，双栈时优先只返回ipv4结果。
    在双栈存在且ipv4可连通的情况下，只返回ipv4结果；在双栈存在且ipv4不可连通ipv6可连通时，只返回ipv6结果；其他只有ipv4或ipv6情况正常返回结果。这样在开启ipv6的情况下网络访问速度和只开启ipv4时基本一直，碰到只有ipv6网址时又可以正常访问，比如家庭宽带的用ddns自建的ipv6网址。mosdns也是有该功能的。

**Comment by PikuZheng** [comment]
重复 #1034 #1194 https://github.com/pymumu/smartdns/issues/1635

类似的也有想要ipv6优先的 #1489 #1585 

如果是时通时不通的 双栈测速和优选满足不了吗？

**Reply by snaking999** [reply]
因为运营商的ipv6会出现阶段性或间断性的不稳定情况，导致dns服务器测试时是好的返回了ipv6地址，但实际经常会有ipv6连不通的情况。运营商目前v6的优化及维护还是有很大问题，还存在区域性的影响，特别我家电信最近1个月v6基本崩了没法使用，关了解析部分v6的网站又没法访问。
可以考虑加入v4及v6优先的功能。
